src/librustc_ast_pretty/pprust.rs

```diff
@@ -1413,8 +1398,29 @@ impl<'a> State<'a> {
     }
 
     crate fn print_assoc_item(&mut self, item: &ast::AssocItem) {
```

Comment:
Hmm; the duplication here and in `visit.rs` isn't great. Let's follow up with the common structs to get some more code reuse.